src/cq_cam/fluent.py

```diff
@@ -36,7 +36,9 @@ def to_gcode(self):
         gcodes = [f"({self.job.name} - {self.name})"]
         previous_command = None
         for command in self.commands:
-            gcode, position = command.to_gcode(previous_command, position)
+            command.previous_command = previous_command
+            command.start = position
+            gcode, position = command.to_gcode()
```

Comment:
Technically this works just fine, but initializing instance attributes outside of `__init__` sets my spider senses tingling. 

It is fairly harmless in this particular small scope, but risks becoming unsustainable if this approach is used liberally throughout the code base. Slippery spaghetti slope :warning: 


I'll think about this still a bit, but I think the appropriate approach might be to make the `to_gcode` more flexible with `**kwargs`. This also applies to the `ConfigCommand` discussion above. Something like this maybe

```python
class BaseCommand(ABC):
    @abstractmethod
    def to_gcode(self, x=None, y=None, z=None):
        pass


class SpecialCommand(BaseCommand):
    def to_gcode(self, x=None, **kwargs):
        # Do something with x
        pass
```

Reply:
With the above approach this line could be something like
```suggestion
            gcode, position = command.to_gcode(operation=self, previous_command=previous_command, start=position)
```

This way it'd be possible to access both the operation and the job (`operation.job`)  in any `to_gcode`  without needing to store copies of top-level state inside Commands.